sdk/core/azure-core/CMakeLists.txt

```
@@ -21,6 +21,14 @@ if(NOT CURL_FOUND)
 endif()
 message("Libcurl version ${CURL_VERSION_STRING}")
 
+# Storage requires 3.6.0 version for using `contains` feature
+set(NLOHMANN_JSON_MIN_REQUIRED_VERSION 3.6.0)
```

nit: Consider removing the comment and updating to latest version.

```suggestion
set(NLOHMANN_JSON_MIN_REQUIRED_VERSION 3.9.1)
```

---

making it 3.9.1 the MIN will force people to update vcpkg to get latest packages (and not sure if 3.9.1 is already published with vcpkg). I think we have to name the version that comes with the vcpkg version we have tagged right?

---

> I think we have to name the version that comes with the vcpkg version we have tagged right?

Yes! Good call.

https://github.com/microsoft/vcpkg/blob/0bf3923f9fab4001c00f0f429682a0853b5749e0/ports/nlohmann-json/CONTROL#L1-L4

---

guess what!  3.9.1 is the one that comes with tag 2020.11, so yes, let's use that one.

---

I agree, no need to request 3.9.1 as minimum vbersion if we're fine with 3.6.0.

---

Unlike libcurl and other dependencies that come  with the underlying OS like Ubuntu 16.04 (which we want to use lowest version for adoption), we should try to use the newest possible library we leverage as part of our implementation, so that we don't carry known bugs.

---

Requiring a min version of 3.6 doesn't mean it is going to install 3.6. For example, if you run `vcpkg install nhollman-json` today, it is going  to install 3.9.1. You need to have a vcpkg cloned and not updated since Mar 25 2019, in order to get exactly 3.6.0. I don't think it is the job of our repo to request the latest version, when we don't really need it, in order to not get bugs that are not in our code, that we are not affected by, and that we can't enumerate, as overall goodness, especially requesting the latest version - it can be an adoption blocker, if someone is using an earlier version and can't update to the latest for cheap.

---

What would prevent someone to upgrade the json library to latest that we'd consider it an adoption blocker? Is the adoption blocker concern you mentioned a general one, or specific to this particular library?

How do you reconcile that with security or other bugs that might show up in the older versions of that library?

---

I am keeping the 3.6.0 as the MIN.
I consider this a more end-user friendly approach. Specially because we now know that any old application using literals like `and` will need to be fixed to work with 3.9.0.
So, at least we can have an option for old applications to still go and install 3.6.0 and then use our SDK without any issue.